src/app/home/panel/cards/transactions/transactions.component.ts

```diff
@@ -47,6 +51,29 @@ export class TransactionsComponent implements OnInit {
     this.transactionsList = this.transactionsList.filter(transaction => transaction.repeat);
   }
 
+  setTransactionsListFilters() {
+    this.panelService.getTransactionsListFilters().subscribe(res => {
+
+      if (res.date && res.date !== '') {
+        this.transactionsList = this.transactionsList
+          .filter(transaction => moment.unix(transaction.date.seconds).format('DD/MM/YYYY') === moment(res.date,).format('DD/MM/YYYY'));
+      }
+
+      if (res.amountFrom && res.amountFrom !== '') {
+        this.transactionsList = this.transactionsList.filter(transaction => transaction.amount > res.amountFrom);
+      }
+
+      if (res.amountTo && res.amountTo !== '') {
+        this.transactionsList = this.transactionsList.filter(transaction => transaction.amount < res.amountTo);
+      }
+
+      if (res.budgets.length > 0) {
+        this.transactionsList = this.transactionsList.filter(transaction => res.budgets.includes(transaction.budgetId));
+      }
+    });
```

Comment:
Try to take advantage of filters callback, and make all the logic there. It will result in only one filtration, instead of four.